prow/config/config.go

```diff
@@ -286,42 +285,26 @@ func Load(prowConfig, jobConfig string) (*Config, error) {
 		return nil, err
 	}
 
-	if !stat.IsDir() {
-		// still support a single file
-		jobConfig, err := yamlToConfig(jobConfig)
-		if err != nil {
-			return nil, err
-		}
-		if err := nc.mergeJobConfig(jobConfig.JobConfig); err != nil {
-			return nil, err
-		}
-		return nc, nil
+	if stat.IsDir() {
+		return nil, fmt.Errorf("jobConfig should be a single file!")
 	}
 
-	err = filepath.Walk(jobConfig, func(path string, info os.FileInfo, err error) error {
-		if err != nil {
-			logrus.WithError(err).Errorf("walking path %q.", path)
-			// bad file should not stop us from parsing the directory
-			return nil
-		}
-
-		if filepath.Ext(path) != ".yaml" {
-			return nil
-		}
+	// split into multiple blobs :-)
+	jobData, err := ioutil.ReadFile(jobConfig)
+	if err != nil {
+		return nil, fmt.Errorf("error reading %s: %v", jobConfig, err)
+	}
 
-		if info.IsDir() {
-			return nil
-		}
+	jobBlobs := strings.Split(string(jobData), "---")
```

Comment:
this isn't quite right. it needs to split on lines that start with `---` at the beginning of the line, `---` is valid in other places like `- foo: "bar --- baz"`

also once split, we need to gracefully handle:
```yaml
---
- foo: bar
   baz: quux
---
- this: is
   fine: doge
---
```
(IE empty ranges)

Reply:
> I would probably do this in a method that generically breaks a byte blob into a slice of yaml-spec-allowed byte slices not-yet deserialized blocks to be deserialized by something aware of the types.

be more specific? Like walk through the file and split by `---\n`? Or reimplement the yaml parser 🤦‍♂️

Reply:
just the part that splits up the documents. walking all _lines)_ and splitting on lines with `---`. (though if we want it to be more reusable, allowing lines with more than 3 `-` to split would be good (not sure if that's standard, but it seems widely used)

Reply:
ack sure I can do that